src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -301,11 +314,12 @@ private async Task<AzureDevOpsClient> GetAzureDevOpsClientForAccount(string acco
             return new AzureDevOpsClient(accessToken, Logger, null);
         }
 
-        private void AddFinishedBuildChannelsIfNotPresent(HashSet<BuildChannel> buildChannelsToAdd)
+        private async Task<List<BuildChannel>> AddFinishedBuildChannelsIfNotPresent(HashSet<BuildChannel> buildChannelsToAdd)
         {
             var missingBuildChannels = buildChannelsToAdd.Where(x => !Context.BuildChannels.Any(y => y.ChannelId == x.ChannelId && y.BuildId == x.BuildId)).ToList();
             Context.BuildChannels.AddRange(missingBuildChannels);
```

Comment:
Should this be AddRangeAsync?

Reply:
AddRangeAsync comes with this remark and it didn't sound like it was applicable?
```
//     This method is async only to allow special value generators, such as the one
//     used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo',
//     to access the database asynchronously. For all other cases the non async method
//     should be used.
```

Reply:
I thought i saw another case where we added with AddRangeAsync (to BuildChannels).

Reply:
We have one AddRangeAsync [here](https://github.com/dotnet/arcade-services/blob/master/src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildsController.cs#L165) 

and there are a lot of AddAsync calls throughout, which also come with the same remark.

I'm not informed enough to know what the downsides of calling it async are exactly, or what exactly constitutes the cases when it should vs it shouldn't be called that way, but I can try and find out more to see if it's something we should be worrying about or if it's fine to keep as is.